lua/nvim-peekup/peekup.lua

```diff
@@ -98,7 +98,7 @@ local function on_keystroke(key)
 	  vim.cmd('redraw')
 	  vim.cmd('sleep '..config.on_keystroke.delay)
 	  vim.cmd('execute "normal! \\<Esc>^"')
-	  vim.cmd('let @*=@'..key)
+	  vim.cmd('let @'..config.paste_reg..'=@'..key)
```

Comment:
I believe here we must check that the string is indeed a valid vim register string, otherwise the assignment will break.

Reply:
Possibly, but OTOH a user's also somewhat responsible for their own overrides. If the documentation says "insert a register name, like `*` or `"` or `a` etc" and the user inserts `#`, well.. they get to keep the pieces when things break.  If there's an easy way to check the register is a valid one, I'd rather that happened at startup though, rather than here.

Reply:
Yes, I believe the check must be done at start-up rather than upon execution. The way to check is essentially to make sure that the chosen register belongs to the object `reg_chars` (which by brute force already lists all available vim registers). Thus we could do (somewhere at initialisation)
```
assert reg_chars[paste_reg]~=nil else reg_chars = '*' (or some default)
```

Reply:
I have implemented a validation check for registers [here](https://github.com/gennaro-tedesco/nvim-peekup/blob/configurable-paste-register/lua/nvim-peekup/config.lua): this branch is basically forked from yours (I am not sure how to refer to forked branches so I created an equivalent one and pushed it 😅 ). There is also a [unit test for the validation function](https://github.com/gennaro-tedesco/nvim-peekup/blob/configurable-paste-register/lua/spec/validate_reg_spec.lua).

Have a look if this does what you intend :)

Reply:
Seems to default a "wrong" register to the `*` register, which may well be good enough. Yup, great!

Reply:
Yes, whenever the register doesn't exist we default to `*` or (anything else we decide). Could you please help me test this behaviour by installing the plugin off this new branch and check that all is good? I have tested it and "it works on my machine" but four eyes are better than two!

If all is good I will merge later.

Reply:
Yup, I have and all looks good. I've _not_ installed the unit testing lua module, though - just checked the vim-related behaviour by adding `lua require('nvim-peekup.config').paste_reg = '"'` to my vimrc and verifying it does what I want as well as adding `lua require('nvim-peekup.config').paste_reg = '$'` (an invalid register) and verifying it modified the `*` register instead.  LGTM with flying colors.